pkg/zeek/type.go

```diff
@@ -316,7 +316,14 @@ func LookupTypeRecord(columns []Column) *TypeRecord {
 	if ok {
 		return t.(*TypeRecord)
 	}
-	rec := &TypeRecord{Columns: columns, Key: s}
+	// Make a private copy of the columns to maintain the invariant
```

Comment:
I'm not sure if there's a lot of contention with all the new mappers etc but if so the copy work could be moved above the `Lock()`. Or maybe do it anyway...

Reply:
You only want the copy if the record isn't in the table and the table check and update needs to be atomic.

Reply:
Yes, but if you get here then the record is  nearly always not in the table, since there was a check (under RLock) above.